src/robot/running/userkeywordrunner.py

```diff
@@ -144,6 +144,19 @@ def _format_trace_log_args_message(self, args, variables):
         args = ['%s=%s' % (name, prepr(variables[name])) for name in args]
         return 'Arguments: [ %s ]' % ' | '.join(args)
 
+    def _inherit_tc_continue_on_failure(self, kw_tags, context):
+        # apply test-case's continue-on-failure on keywords
+        # (unless robot:no-continue-on-failure is set in keyword)
+        try:
+            tc_tags = context.test.tags
+        except AttributeError:
+            return kw_tags
+        result = list(kw_tags)
+        if ('robot:continue-on-failure' in tc_tags
+           and 'robot:no-continue-on-failure' not in kw_tags):
+            result.append('robot:continue-on-failure')
+        return result
```

Comment:
Do I get it right that the idea of this function is to get the default value for the continuation functionality from the test case? That sounds good, but shouldn't we also look at possible parent user keywords? I just looked at the issue #2285 and it seems that we haven't yet agreed on [this part of the design](https://github.com/robotframework/robotframework/issues/2285#issuecomment-472770743). Should we discuss that further on Slack?

If we want to look at the parent keywords, we have a problem they may not be easily accessible at the moment. It ought to be pretty easy to change the current context object to preserve them, though.

This function returns tags as a list, not as a `Tags` object, and thus `tag in tags` check isn't anymore case and space insensitive. Instead of returning tags in any container, perhaps this method should just return `True` or `False` depending on is continue-on-failure enabled or not? That would then require changes everywhere else as well, but the logic would probably get simpler. Case and space insensitivity with `robot:continue-on-failure` should also be tested.

Perhaps we should actually move checking is continue-on-failure enabled fully to the context object? It already has access to the current test and, if needed, could also get access to parent user keywords. It could thus just have a property like `continue_on_failure` or `continue_on_failure_mode` that returns `True/False`. Then we could simply pass that value to `ExecutionStatus.can_continue` instead of tags.

Reply:
removed this code